examples/basic/src/index.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { useRef } from "react";
+import { createRoot } from "react-dom/client";
+import { Typography } from "antd";
+import { useEffect, useState } from "react";
+import type { EmbeddedSearch } from "@gleanwork/web-sdk";
+
+import "./styles.css";
+
+declare global {
+  interface Window {
+    EmbeddedSearch?: EmbeddedSearch;
+  }
+}
+
+function App() {
+  const containerRef = useRef(null);
+  const [isGleanReady, setIsGleanReady] = useState(false);
+
+  // initialize glean
+  useEffect(() => {
+    if (!isGleanReady || !containerRef.current) return;
+
+    window.EmbeddedSearch?.attach(containerRef.current, {
+      onSearch(query) {
+        console.log(query);
+      },
+    });
+  }, [isGleanReady]);
+
+  // load glean script and mark glean as ready once its loaded
+  useEffect(() => {
+    const script = document.createElement("script");
+    script.src = "https://app.glean.com/embedded-search-latest.min.js";
+    script.onload = () => {
```

Comment:
Main fix to prevent "invoke before script load" race conditions.

Gives other effects "observability" on whether Glean has been initialized